File: CatalogueMovieSubmission2/app/src/main/java/com/azhara/cataloguemoviesubmission2/Fragment/MoviesFragment.java
package com.azhara.cataloguemoviesubmission2.Fragment;


import android.content.Intent;
import android.content.res.TypedArray;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.azhara.cataloguemoviesubmission2.Adapter.MovieAdapter;
import com.azhara.cataloguemoviesubmission2.DetailActivity;
import com.azhara.cataloguemoviesubmission2.Model.MovieTvShow;
import com.azhara.cataloguemoviesubmission2.R;

import java.util.ArrayList;

/**
 * A simple {@link Fragment} subclass.
 */
public class MoviesFragment extends Fragment {
    private TypedArray photo;
    private String[] judul;
    private String[] sinopsis;
    private String[] durasi;
    private String[] genre;
    private String[] rilis;
    private RecyclerView rvMovie;
    private ArrayList<MovieTvShow> list = new ArrayList<>();


    public MoviesFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return  inflater.inflate(R.layout.fragment_movies, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        prepareDataFilm();
        rvMovie = view.findViewById(R.id.rv_movies);
        rvMovie.setHasFixedSize(true);
        list.addAll(setItem());
        init();

    }

    private void init(){
        rvMovie.setLayoutManager(new LinearLayoutManager(getContext()));
        MovieAdapter movieAdapter = new MovieAdapter(list);
        rvMovie.setAdapter(movieAdapter);

        movieAdapter.setOnItemClickCallBack(new MovieAdapter.OnItemClickCallBack() {
            @Override
            public void onItemClicked(MovieTvShow movies) {
                detailMovie(movies);
            }
        });
    }

    private void prepareDataFilm(){
        photo = getResources().obtainTypedArray(R.array.photo_film);
        judul = getResources().getStringArray(R.array.judul_film);
        sinopsis = getResources().getStringArray(R.array.sinopsis_film);
        durasi = getResources().getStringArray(R.array.durasi_film);
        genre = getResources().getStringArray(R.array.genre_film);
        rilis = getResources().getStringArray(R.array.release_film);
    }

    private ArrayList<MovieTvShow> setItem(){
        ArrayList<MovieTvShow> listmovies = new ArrayList<>();
        for (int i = 0; i < judul.length; i++){
            MovieTvShow movieTvShow = new MovieTvShow();
            movieTvShow.setPicture(photo.getResourceId(i,-1));
            movieTvShow.setJudul(judul[i]);
            movieTvShow.setSinopsis(sinopsis[i]);
            movieTvShow.setDurasi(durasi[i]);
            movieTvShow.setGenre(genre[i]);
            movieTvShow.setRelease(rilis[i]);
            listmovies.add(movieTvShow);
        }

        return listmovies;
    }

    private void detailMovie(MovieTvShow movieTvShow){
        MovieTvShow movies = new MovieTvShow();
        movies.setPicture(movieTvShow.getPicture());
        movies.setJudul(movieTvShow.getJudul());
        movies.setSinopsis(movieTvShow.getSinopsis());
        movies.setDurasi(movieTvShow.getDurasi());
        movies.setGenre(movieTvShow.getGenre());
        movies.setRelease(movieTvShow.getRelease());

        Intent intent = new Intent(getActivity(), DetailActivity.class);
        intent.putExtra(DetailActivity.DETAIL_FILM, movies);
        startActivity(intent);

    }
}

File: ProyekAkhirDicoding2/app/src/main/java/com/azhara/proyekakhirdicoding/ui/movies/MoviesFragment.kt
package com.azhara.proyekakhirdicoding.ui.movies


import android.annotation.SuppressLint
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.azhara.proyekakhirdicoding.R
import com.azhara.proyekakhirdicoding.ui.movies.adapter.MoviesAdapter
import com.azhara.proyekakhirdicoding.ui.movies.model.MoviesModel
import com.azhara.proyekakhirdicoding.ui.movies.viewmodel.MoviesViewModel
import kotlinx.android.synthetic.main.fragment_movies.*

/**
 * A simple [Fragment] subclass.
 */
class MoviesFragment : Fragment() {

    private lateinit var moviesViewModel: MoviesViewModel
    private lateinit var moviesAdapter: MoviesAdapter

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_movies, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        shimmer_movies.startShimmer()

        moviesAdapter = MoviesAdapter()
        moviesAdapter.notifyDataSetChanged()
        rv_movies.layoutManager = LinearLayoutManager(activity)
        rv_movies.adapter = moviesAdapter

        setData()
        getData()

        moviesAdapter.setOnItemClickCallBack(object : MoviesAdapter.OnItemClickCallBack {
            override fun onItemClicked(data: MoviesModel) {
                toDetail(data)
            }

        })

    }

    private fun toDetail(moviesData: MoviesModel) {
        val movies = MoviesModel()
        val posterPathUrl = "https://image.tmdb.org/t/p/w342${moviesData.poster}"
        val backgroundUrl = "https://image.tmdb.org/t/p/original${moviesData.background}"

        movies.title = moviesData.title
        movies.poster = posterPathUrl
        movies.rating = moviesData.rating
        movies.background = backgroundUrl
        movies.vote = moviesData.vote
        movies.release = moviesData.release
        movies.overview = moviesData.overview

        val intent = Intent(activity, DetailMoviesActivity::class.java)
        intent.putExtra(DetailMoviesActivity.EXTRA_DATA, movies)
        startActivity(intent)

    }

    private fun setData() {
        moviesViewModel = ViewModelProvider(
            this,
            ViewModelProvider.NewInstanceFactory()
        ).get(MoviesViewModel::class.java)
        moviesViewModel.setData()
    }

    @SuppressLint("FragmentLive DataObserve")
    private fun getData() {
        moviesViewModel.getData().observe(viewLifecycleOwner, Observer { moviesData ->
            moviesAdapter.setData(moviesData)
            if (shimmer_movies.visibility == View.VISIBLE) {
                shimmer_movies.stopShimmer()
                shimmer_movies.visibility = View.GONE
            }
        })
    }

}
